html/dom/documents/dom-tree-accessors/nameditem-fetch-01.html

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Named items: fetch text/html</title>
+<link rel="author" title="bzbarsky" href="mailto:bzbarsky@mit.edu">
+<link rel="author" title="ExE Boss" href="https://ExE-Boss.tech">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#dom-htmldocument-nameditem">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+test(function() {
+  var url = 'data:text/html,<html xmlns="http://www.w3.org/1999/xhtml"><form name="x"/></html>';
+  var xhr = new XMLHttpRequest();
+  xhr.open("GET", url);
+  xhr.responseType = "document";
+  xhr.onload = function() {
+    var doc = xhr.responseXML;
```

Comment:
I don't think it's defined anywhere currently (or in a proposal) that this returns an `HTMLDocument` instance.

Reply:
For&nbsp;`text/html`, which&nbsp;this&nbsp;is.

The&nbsp;other&nbsp;test performs&nbsp;fetch for&nbsp;`application/xml`, that&nbsp;returns&nbsp;`Document` in&nbsp;all&nbsp;browsers tested&nbsp;by&nbsp;me.

Reply:
`text/html` is an internal concept of the object. It doesn't translate into returning a different class. I'm talking about the specification here, not about what implementations do.

Reply:
Hmm.  So in terms of the specification, the bits in https://github.com/whatwg/html/pull/5104 would need to say when an HTMLDocument should be created, right?